ui/mainwindow.cpp

```diff
@@ -111,9 +121,9 @@ MainWindow::MainWindow(QWidget *parent): QMainWindow(parent), ui(new Ui::MainWin
 
   setWindowTitle(tr("GammaRay (%1)").arg(Endpoint::instance()->label()));
 
-  selectInitialTool();
```

Comment:
No, that is not what I meant, and it is actually a simpler problem.
If I understand correctly the initial tool is supposed to be selected exactly once. There are two scenarios:
 1. The `ui->toolSelector->model()` is fully populated in the constructor. I this case the initial tool is selected by the constructor.
 2. There is no fitting tool in the model. I this case the initial tool is selected, when first a fitting tool is inserted into the model. Thats why the `rowsInserted` and the `dataChanged` signal is connected to the `selectInitialTool` slot.

But if it is so, these signals should not remain connected after the `selectInitialTool` has run succesfully in the constructor, thats why I suspect that the order of these lines are intended to be as I wrote them.

Would be much cleaner, if these signal would only be connected if and when it is deteced that there are no tool to select initially.